tests/unit/server/bundle_manager/make_bundles_test.py

```diff
@@ -0,0 +1,150 @@
+
+    def test_multiple_dependencies(self):
+        """A MakeBundle with two dependencies should be made."""
+        bundle, parent1, parent2 = self.create_bundle_two_deps()
+        self.save_bundle(bundle)
+        self.save_bundle(parent1)
+        self.save_bundle(parent2)
+
+        threads = self.bundle_manager._make_bundles()
+        self.assertTrue(self.bundle_manager._is_making_bundles())
+        for t in threads:
+            t.join()
+        self.assertFalse(self.bundle_manager._is_making_bundles())
+
+        bundle = self.bundle_manager._model.get_bundle(bundle.uuid)
+
+        self.assertEqual(bundle.state, State.READY)
+
+        with self.read_bundle(bundle, "src1") as f:
+            self.assertEqual(f.read(), "hello world 1")
+        with self.read_bundle(bundle, "src2") as f:
+            self.assertEqual(f.read(), "hello world 2")
+
+    def test_fail_invalid_dependency_path(self):
+        """A MakeBundle with an invalid dependency specified should fail."""
+        bundle = self.create_make_bundle(state=State.STAGED)
+        bundle.dependencies = [
+            Dependency(
+                {
+                    "parent_uuid": generate_uuid(),
+                    "parent_path": "",
+                    "child_uuid": bundle.uuid,
+                    "child_path": "src",
+                }
+            )
+        ]
+        self.save_bundle(bundle)
+
+        threads = self.bundle_manager._make_bundles()
+        self.assertTrue(self.bundle_manager._is_making_bundles())
+        for t in threads:
+            t.join()
+        self.assertFalse(self.bundle_manager._is_making_bundles())
+
+        bundle = self.bundle_manager._model.get_bundle(bundle.uuid)
+
+        self.assertEqual(bundle.state, State.FAILED)
+        self.assertIn("Invalid dependency", bundle.metadata.failure_message)
+
+    def test_linked_dependency(self):
+        """A MakeBundle with a linked dependency should be made."""
+        with tempfile.NamedTemporaryFile(delete=False) as f:
+            f.write(b"hello world")
+            tempfile_name = f.name
+        parent = DatasetBundle.construct(
+            metadata=dict(BASE_METADATA_DATASET_BUNDLE, link_url=tempfile_name),
+            owner_id=self.user_id,
+            uuid=generate_uuid(),
+        )
+        bundle = self.create_make_bundle(state=State.STAGED)
+        bundle.dependencies = [
+            Dependency(
+                {
+                    "parent_uuid": parent.uuid,
+                    "parent_path": "",
+                    "child_uuid": bundle.uuid,
+                    "child_path": "src",
+                }
+            )
+        ]
+        self.save_bundle(parent)
+        self.save_bundle(bundle)
+
+        threads = self.bundle_manager._make_bundles()
+        self.assertTrue(self.bundle_manager._is_making_bundles())
+        for t in threads:
+            t.join()
+        self.assertFalse(self.bundle_manager._is_making_bundles())
```

Comment:
Could we refactor this piece that shows up earlier?

Reply:
Done